src/zope/schema/interfaces.py

```diff
@@ -387,13 +535,23 @@ class IInt(IMinMax, IField):
         )
 
 
-class IFloat(IMinMax, IField):
-    """Field containing a Float."""
+class IFloat(IReal):
+    """
+    Field containing exactly the native class :class:`float`.
+
+    :class:`IReal` is a more general interface, allowing all of
+    floats, ints, and fractions.
+
+    .. seealso:: :class:`zope.schema.Float`
+    """
 
 
-class IDecimal(IMinMax, IField):
-    """Field containing a Decimal."""
+class IDecimal(INumber):
+    """Field containing a :class:`decimal.Decimal`"""
```

Comment:
`Decimal` is just an `Number`, so the fields `IDecimal` inherits from `INumber` are appropriate.

(It turns out it doesn't really matter, though, the values supplied to the constructor are checked against the field's `_type`, which is `decimal.Decimal`, so that's all you can actually use for a `Decimal` field.)